generation/internal/node.go

```diff
@@ -65,12 +65,26 @@ func nodeTestJob(ls labels.LabelSet) *Job {
 
 	steps := []config.Step{
 		checkoutStep(ls[labels.DepsNode]),
-		{
-			Type:       config.OrbCommand,
-			Command:    "node/install-packages",
-			Parameters: config.OrbCommandParameters{"pkg-manager": packageManager},
-		},
 	}
+
+	if ls[labels.DepsNode].HasLockFile {
+		steps = append(steps, config.Step{
+			Type:    config.OrbCommand,
+			Command: "node/install-packages",
+			Parameters: config.OrbCommandParameters{
+				"pkg-manager": packageManager,
+			}})
+	} else {
+		// override the default ci command to not use the lockfile
+		steps = append(steps, config.Step{
+			Type:    config.OrbCommand,
```

Comment:
~ wondering if we should add a comment to the step

Reply:
Added! [`2bc1e85` (#33)](https://github.com/CircleCI-Public/circleci-config/pull/33/commits/2bc1e85791e59a8e6fab925f05fa3ce49f571343)